ledger/voters.go

```diff
@@ -192,6 +197,35 @@ func (vt *votersTracker) newBlock(hdr bookkeeping.BlockHeader) {
 
 }
 
+func (vt *votersTracker) prepareCommit(dcc *deferredCommitContext) error {
+	vt.commitListenerMu.RLock()
+	defer vt.commitListenerMu.RUnlock()
+
+	if vt.commitListener == nil {
+		return nil
+	}
+
+	commitListener := *vt.commitListener
+	for round := dcc.oldBase; round <= dcc.newBase; round++ {
+		err := commitListener.OnPrepareVoterCommit(round, vt)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (vt *votersTracker) postCommit(dcc *deferredCommitContext) {
+	lastHeaderCommitted, err := vt.l.BlockHdr(dcc.newBase)
+	if err != nil {
+		vt.l.trackerLog().Errorf("votersTracker.postCommit: could not retrieve header for round %d: %v", dcc.newBase, err)
+		return
+	}
+
+	vt.removeOldVoters(lastHeaderCommitted)
```

Comment:
the code in the function is called under two locks : parent's onlineact lock and `vt.votersMu.Lock()` below

Reply:
That is correct. Is that problematic?

Reply:
Fixed, take a look now